rust/src/x509/mod.rs

```diff
@@ -79,6 +94,34 @@ pub unsafe extern "C" fn rs_x509_get_subject(ptr: *const X509) -> *mut c_char {
     rust_string_to_c(subject)
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn rs_x509_get_subjectaltname_len(ptr: *const X509) -> u16 {
+    if ptr.is_null() {
+        return 0;
+    }
+    let x509 = cast_pointer! {ptr, X509};
+    let san_list = x509.0.tbs_certificate.subject_alternative_name();
+    if let Ok(Some(sans)) = san_list {
+        return sans.value.general_names.len() as u16;
```

Comment:
How are we sure sans.value.general_names has less than U16_MAX ?

Reply:
I used `u16` by searching at various places about the max SANs allowed in a certificate. All said different things but I think I relied on this thread: https://community.letsencrypt.org/t/why-sans-are-limited-to-100-domains-only/154930
This told me that:
1. LetsEncrypt allows 100 SANs in a certificate to avoid being seen doing a DoS attack on the authoritative DNS Servers
2. People can make at least up to 10k SANs whether they're correctly handled or not.

So, to be on a safe side, I chose `u16`, wdyt?

Reply:
Ok, could you add that reference as a comment ?
And maybe a debug_validation that sans.value.general_names.len() <= U16_MAX ?